### PR TITLE
Added changes for Byopip feature

### PR DESCRIPTION
**Pull request by udit1402**
## Description
Issue link: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1414
So this Byopip is a new feature getting added, and it doesn't include any new Commandlet. Just changing the exisiting Commandlet with the proper validations

## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Network/Network/AzureFirewall/NewAzureFirewallCommand.cs [hunk 1]
```diff
@@ -310,11 +319,16 @@ private PSAzureFirewall CreateAzureFirewall()
                     Sku = sku,
                     VirtualHub = VirtualHubId != null ? new MNM.SubResource(VirtualHubId) : null,
                     FirewallPolicy = FirewallPolicyId != null ? new MNM.SubResource(FirewallPolicyId) : null,
-                    HubIPAddresses = this.HubIPAddress,
+                    HubIPAddresses = this.HubIPAddress == null ? null : this.HubIPAddress,
```
Comment (vidai-msft) [inline src/Network/Network/AzureFirewall/NewAzureFirewallCommand.cs:322]: Is this the same?

File: src/Network/Network/Models/AzureFirewall/PSAzureFirewall.cs [hunk 2]
```diff
@@ -386,6 +386,24 @@ public void RemovePublicIpAddress(PSPublicIpAddress publicIpAddress)
             }
         }
 
+        public void AddIpAddressesForByopipHubFirewall(PSPublicIpAddress[] publicIpAddresses)
+        {
+            this.IpConfigurations = new List<PSAzureFirewallIpConfiguration>();
+
+            if (publicIpAddresses != null && publicIpAddresses.Count() > 0)
+            {
+                for (var i = 0; i < publicIpAddresses.Count(); i++)
```
Comment (vidai-msft) [inline src/Network/Network/Models/AzureFirewall/PSAzureFirewall.cs:395]: Count or Length is preferable over Count() as this is an extension method from Linq which may cause performance overhead.

File: src/Network/Network/Models/AzureFirewall/PSAzureFirewall.cs [hunk 3]
```diff
@@ -386,6 +386,24 @@ public void RemovePublicIpAddress(PSPublicIpAddress publicIpAddress)
...
+                {
+                    this.IpConfigurations.Add(
+                        new PSAzureFirewallIpConfiguration
+                        {
+                            Name = $"{AzureFirewallIpConfigurationName}{i}",
+                            PublicIpAddress = new PSResourceId { Id = publicIpAddresses[i].Id }
```
Comment (vidai-msft) [inline src/Network/Network/Models/AzureFirewall/PSAzureFirewall.cs:401]: Is null check necessary here?

**Review by vidai-msft — CHANGES_REQUESTED** [review]

**Comment by vidai-msft** [comment]
@udit1402 Please remember to update `ChangeLog.md`. This is required for publishing the code.

File: src/Network/Network/Models/AzureFirewall/PSAzureFirewall.cs (hunk 3 above)
Comment (udit1402) [reply src/Network/Network/Models/AzureFirewall/PSAzureFirewall.cs:401]: Not necessarily. If you'd see the function from where it is called, I am already doing a null check on the object

File: src/Network/Network/AzureFirewall/NewAzureFirewallCommand.cs (hunk 1 above)
Comment (udit1402) [reply src/Network/Network/AzureFirewall/NewAzureFirewallCommand.cs]: Yea that's right, updated back

**Review by vidai-msft — APPROVED** [review]
LGTM

**Pull request merged**